resilience4j-micrometer/src/main/java/io/github/resilience4j/micrometer/Timer.java

```diff
@@ -0,0 +1,428 @@
+     *
+     * @param name     the name of the Timer
+     * @param registry the registry to bind Timer to
+     * @return a Timer with a custom Timer configuration.
+     */
+    static Timer of(String name, MeterRegistry registry) {
+        return of(name, registry, TimerConfig.ofDefaults(), emptyMap());
+    }
+
+    /**
+     * Creates a Timer with a custom Timer configuration.
+     *
+     * @param name        the name of the Timer
+     * @param registry    the registry to bind Timer to
+     * @param timerConfig a custom Timer configuration
+     * @return a Timer with a custom Timer configuration.
+     */
+    static Timer of(String name, MeterRegistry registry, TimerConfig timerConfig) {
+        return of(name, registry, timerConfig, emptyMap());
+    }
+
+    /**
+     * Creates a Timer with a custom Timer configuration.
+     *
+     * @param name        the name of the Timer
+     * @param registry    the registry to bind Timer to
+     * @param timerConfig a custom Timer configuration
+     * @param tags        tags to assign to the Timer
+     * @return a Timer with a custom Timer configuration.
+     */
+    static Timer of(String name, MeterRegistry registry, TimerConfig timerConfig, Map<String, String> tags) {
+        return new TimerImpl(name, registry, timerConfig, tags);
+    }
+
+    /**
+     * Decorates CompletionStageSupplier with Timer
+     *
+     * @param timer    the timer
+     * @param supplier completion stage supplier
+     * @param <T>      type of completion stage result
+     * @return decorated supplier
+     */
+    static <T> Supplier<CompletionStage<T>> decorateCompletionStage(Timer timer, Supplier<CompletionStage<T>> supplier) {
+        return () -> {
+            CompletableFuture<T> promise = new CompletableFuture<>();
+            Context context = timer.createContext();
+            try {
+                supplier.get().whenComplete((output, throwable) -> {
+                    if (throwable != null) {
+                        context.onFailure(throwable);
+                        promise.completeExceptionally(throwable);
+                    } else {
+                        context.onSuccess(output);
+                        promise.complete(output);
+                    }
+                });
+            } catch (Exception e) {
+                context.onFailure(e);
+                promise.completeExceptionally(e);
+            }
+            return promise;
+        };
+    }
+
+    /**
+     * Decorates CheckedSupplier with Timer
+     *
+     * @param timer    the timer
+     * @param supplier the supplier
+     * @param <T>      the supplier result
+     * @return decorated supplier
+     */
+    static <T> CheckedSupplier<T> decorateCheckedSupplier(Timer timer, CheckedSupplier<T> supplier) {
+        return () -> {
+            Context context = timer.createContext();
+            try {
+                T output = supplier.get();
+                context.onSuccess(output);
```

Comment:
Let's rename `onSuccess` to `onResult` to make it analog to CircuitBreaker or Retry

Reply:
Done